plugins/flytekit-airflow/flytekitplugins/airflow/agent.py

```diff
@@ -1,108 +1,145 @@
+            except TaskDeferred as td:
+                parameters = td.trigger.__dict__.copy()
+                # Remove parameters that are in the base class
+                parameters.pop("task_instance", None)
+                parameters.pop("trigger_id", None)
+
+                resource_meta.airflow_trigger = AirflowObj(
+                    module=td.trigger.__module__, name=td.trigger.__class__.__name__, parameters=parameters
+                )
+                resource_meta.airflow_trigger_callback = td.method_name
 
         return CreateTaskResponse(resource_meta=cloudpickle.dumps(resource_meta))
 
-    def get(self, context: grpc.ServicerContext, resource_meta: bytes) -> GetTaskResponse:
+    async def async_get(self, context: grpc.ServicerContext, resource_meta: bytes) -> GetTaskResponse:
         meta = cloudpickle.loads(resource_meta)
-        airflow_config = meta.airflow_config
-        job_id = meta.job_id
-        task = _get_airflow_task(FlyteContextManager.current_context(), meta.airflow_config)
+        airflow_operator_instance = _get_airflow_instance(meta.airflow_operator)
+        airflow_trigger_instance = _get_airflow_instance(meta.airflow_trigger) if meta.airflow_trigger else None
+        airflow_ctx = Context()
+        message = None
         cur_state = RUNNING
 
-        if issubclass(type(task), BaseSensorOperator):
-            if task.poke(context=Context()):
+        if isinstance(airflow_operator_instance, BaseSensorOperator):
+            ok = airflow_operator_instance.poke(context=airflow_ctx)
+            cur_state = SUCCEEDED if ok else RUNNING
+        elif isinstance(airflow_operator_instance, BaseOperator):
+            if airflow_trigger_instance:
+                try:
+                    # Airflow trigger returns immediately when
+                    # 1. Failed to get the task status
+                    # 2. Task succeeded or failed
+                    # succeeded or failed: returns a TriggerEvent with payload
+                    # running: runs forever, so set a default timeout (2 seconds) here.
+                    # failed to get the status: raises AirflowException
+                    event = await asyncio.wait_for(airflow_trigger_instance.run().__anext__(), 2)
+                    try:
+                        # Trigger callback will check the status of the task in the payload, and raise AirflowException if failed.
+                        trigger_callback = getattr(airflow_operator_instance, meta.airflow_trigger_callback)
+                        trigger_callback(context=airflow_ctx, event=typing.cast(TriggerEvent, event).payload)
+                        cur_state = SUCCEEDED
+                    except AirflowException as e:
+                        cur_state = RETRYABLE_FAILURE
+                        message = e.__str__()
+                except asyncio.TimeoutError:
+                    logger.debug("No event received from airflow trigger")
```

Comment:
why we don't capture this timeout in `cur_state`?

Reply:
if no event received from airflow trigger, it means job still running. Default state here is running